mlflow/protos/databricks_model_artifacts.proto

```diff
@@ -0,0 +1,89 @@
+syntax = "proto2";
+
+package mlflow;
+
+import "scalapb/scalapb.proto";
+import "databricks.proto";
+import "service.proto";
+
+option java_package = "com.databricks.api.proto.mlflow";
+option java_generate_equals_and_hash = true;
+option py_generic_services = true;
+option (scalapb.options) = {
+  flat_package: true,
+};
+
+service DatabricksModelArtifactsService {
+
+  // Fetch the pre-signed url to read from the specified model registry artifact location
+  //
+  // Note: Even if no artifacts exist at the specified artifact location, this API will
+  // still provide read credentials as long as the format of the location is valid.
+  // Callers must subsequently check for the existence of the artifacts using the appropriate
+  // cloud storage APIs
+  rpc getModelVersionSignedDownloadUri (GetModelVersionSignedDownloadUri) returns (GetModelVersionSignedDownloadUri.Response) {
+    option (rpc) = {
+      endpoints: [{
+        method: "GET",
+        path: "/mlflow/model-versions/get-signed-download-uri"
+        since { major: 2, minor: 0 },
+      }],
+      visibility: PUBLIC_UNDOCUMENTED,
+    };
+  }
+
+  // List model artifacts for a model saved in the model registry.
+  rpc listModelArtifacts (ListModelArtifacts) returns (ListModelArtifacts.Response) {
+    option (rpc) = {
+      endpoints: [{
+        method: "GET",
+        path: "/mlflow/model-versions/list-artifacts"
+        since { major: 2, minor: 0 },
+      }],
+      visibility: PUBLIC_UNDOCUMENTED,
+    };
+  }
+}
+
+message GetModelVersionSignedDownloadUri {
+  option (scalapb.message).extends = "com.databricks.rpc.RPC[$this.Response]";
+
+  // Name of the registered model
+  optional string name = 1 [(validate_required) = true];
+
+  // Model version number
+  optional string version = 2 [(validate_required) = true];
+
+  // The artifact path, relative to the model artifact's root location, for which to
+  // fetch artifact read credentials
+  optional string path = 3 [(validate_required) = true];
+
+  message Response {
+    // A pre-signed url to download a model artifact corresponding to the requested path
+    optional string artifact_uri = 1;
+  }
+}
+
+message ListModelArtifacts {
+  option (scalapb.message).extends = "com.databricks.rpc.RPC[$this.Response]";
+
+  // Name of the registered model
+  optional string name = 1 [(validate_required) = true];
+
+  // Model version number
+  optional string version = 2 [(validate_required) = true];
+
+  // Filter artifacts matching this path (a relative path from the model's root artifact directory).
+  optional string path = 3;
```

Comment:
What is the default behavior if this parameter is left empty? Could you make that clear in the docs?

Reply:
This is optional since it is not required for the request to actually pass validation (be received by the MLflow service).
However, if an empty path is passed, then a 4xx code is returned saying 
`response: {"error_code":"INVALID_PARAMETER_VALUE","message":"Missing required field: path"}`

Reply:
Got it - why not mark it as `validate_required` then?

Reply:
Makes sense, will do! Thanks

Reply:
Ohh wait, Sorry, I thought you were pointing towards the GetModelVersionSignedDownloadUri proto.
For list model artifacts, it is perfectly okay to not pass in the path, hence the optional. In fact, the `download_artifacts` will aim to list the artifacts without passing in the path.
When a call is made without a `path`, then the API returns a list of FileInfos from the root location ,i.e., from `dbfs:/databricks/mlflow-registry/<model-version-id>/models/<path-name>`.